source/simulator/helper_functions.cc

```diff
@@ -2275,13 +2275,28 @@ namespace aspect
   {
     const Quadrature<dim-1> &quadrature_formula = introspection.face_quadratures.temperature;
 
+    bool consider_darcy_velocity = false;
+    unsigned int porosity_idx = numbers::invalid_unsigned_int;
+    if (introspection.composition_type_exists(CompositionalFieldDescription::porosity))
+      {
+        porosity_idx = introspection.find_composition_type(CompositionalFieldDescription::porosity);
+        if (introspection.compositional_field_methods[porosity_idx] == Parameters<dim>::AdvectionFieldMethod::fem_darcy_field)
+          consider_darcy_velocity = true;
+      }
+
     FEFaceValues<dim> fe_face_values (*mapping,
                                       finite_element,
                                       quadrature_formula,
                                       update_values   | update_normal_vectors |
                                       update_quadrature_points | update_JxW_values);
 
     std::vector<Tensor<1,dim>> face_current_velocity_values (fe_face_values.n_quadrature_points);
+    std::vector<Tensor<1,dim>> face_current_fluid_velocity_values (fe_face_values.n_quadrature_points);
+
+    MaterialModel::MaterialModelInputs<dim> in(fe_face_values.n_quadrature_points, introspection.n_compositional_fields);
+    MaterialModel::MaterialModelOutputs<dim> out(fe_face_values.n_quadrature_points, introspection.n_compositional_fields);
+    MeltHandler<dim>::create_material_model_outputs(out);
+    MaterialModel::MeltOutputs<dim> *fluid_out = out.template get_additional_output<MaterialModel::MeltOutputs<dim>>();;
```

Comment:
You are using the material model outputs below, but you do not evaluate the material model. So I don't think the values will be right.

Reply:
To call evaluate, you first need to call `in.reinit(...)`, and you need to do that within the loop over all faces (below line 2326). Each material model inputs object has only as many points as there are on one face of one mesh cell, so each time you go to a new face, you need to update the values. Then once you have the right values in the material model inputs, you can call evaluate to get the right values in the outputs. Then you can use them.